### PR TITLE
Revendor gardener/gardener@v1.11

### PR DESCRIPTION
**Pull request by prashanth26**
**How to categorize this PR?**

/area delivery
/kind enhancement
/priority normal
/platform gcp

**What this PR does / why we need it**:
Revendors the latest gardener/gardener vendors. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Code changes are inspired by https://github.com/gardener/gardener-extension-provider-openstack/pull/155. Credits: @kon-angelo 

**Release note**:

```improvement operator
`github.com/gardener/gardener` dependency is now updated to `v1.11.1`.
```

### CODE REVIEW

**Comment by prashanth26** [comment]
Hi,

I have tried to revendor g/g@1.11. However, when i locally run `make test` i am facing the following issue with this PR
```
fatal error: checkptr: converted pointer straddles multiple allocations

goroutine 1 [running, locked to thread]:
runtime.throw(0x32501c5, 0x3a)
	/usr/local/opt/go/libexec/src/runtime/panic.go:1116 +0x72 fp=0xc0004986c8 sp=0xc000498698 pc=0x1079d72
runtime.checkptrAlignment(0xc0000abbf8, 0x310f7c0, 0x1)
	/usr/local/opt/go/libexec/src/runtime/checkptr.go:20 +0xc9 fp=0xc0004986f8 sp=0xc0004986c8 pc=0x1049569
github.com/karrick/godirwalk.readDirents(0xc0005b5fb0, 0x2b, 0xc0000ab000, 0x1000, 0x1000, 0x0, 0xc000000180, 0xc0004988c8, 0x10aeea8, 0xc000000180)
	/Users/i341888/Go/src/github.com/gardener/gardener-extension-provider-gcp/vendor/github.com/karrick/godirwalk/readdir_unix.go:51 +0x18c fp=0xc000498860 sp=0xc0004986f8 pc=0x2db9b6c
github.com/karrick/godirwalk.ReadDirents(...)
```

However, when I run the same without the `-race` option in the `go test` it works well. I tried to debug to find out the root cause, however, I was unable to find the root cause. Any ideas?

cc: @kon-angelo

**Comment by ialidzhikov** [comment]
See https://github.com/gobuffalo/packr/issues/261. For me it works fine using `make test` in docker container

```
$ docker run --rm -v ${GOPATH}/src:/go/src -it golang:1.15.2 bash
# cd src/github.com/gardener/gardener-extension-provider-gcp
# make test
```

**Comment by kon-angelo** [comment]
That is exactly the issue I had with my local setup too.

**Comment by prashanth26** [comment]
@ialidzhikov - I got a hint that it was an issue with my underlying setup. I tried upgrading my golang but didn't help. However, this issue explains it. Thanks for the hint. Saved my time 👍 

@kon-angelo - Yes, I remember you vaguely mentioning this in the meeting and hence tagged you. Thank you as well 👍

**Review by kon-angelo — APPROVED** [review]
/lgtm

Because our PRs are similar you could wait for a *third* opinion too :)

**Review by ialidzhikov — CHANGES_REQUESTED** [review]
Can you add "SKIP_FETCH_TOOLS=1" to make test?

**Comment by prashanth26** [comment]
> Can you add "SKIP_FETCH_TOOLS=1" to make test?

I have made the suggested changes here - https://github.com/gardener/gardener-extension-provider-gcp/pull/183/commits/d8d5113d3d00917e9156a845e3b63d16311aa5d8. 

Also, this commit - https://github.com/gardener/gardener-extension-provider-gcp/pull/183/commits/ef46c250c30cbb1a4fe942c18d91c26e1cfe3bd7 was forced pushed as a reworded the commit message to capitalize.

File: Makefile [hunk 1]
```diff
@@ -127,7 +127,7 @@ format:
 
 .PHONY: test
 test:
-	@$(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/test.sh ./cmd/... ./pkg/...
+	@SKIP_FETCH_TOOLS=1 $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/test.sh ./cmd/... ./pkg/...
```
Comment (ialidzhikov) [inline Makefile:130]: Can you add it also to `test-cov`?

File: pkg/controller/worker/machines_test.go [hunk 2]
```diff
@@ -549,11 +550,24 @@ var _ = Describe("Machines", func() {
...
 						},
-					}))
+					}
+
+					workerWithExpectedImages := w.DeepCopy()
+					workerWithExpectedImages.Status.ProviderStatus = &runtime.RawExtension{
+						Object: expectedImages,
+					}
+					ctx := context.TODO()
+					c.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&extensionsv1alpha1.Worker{})).
```
Comment (ialidzhikov) [inline pkg/controller/worker/machines_test.go:560]:
```suggestion
					c.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&extensionsv1alpha1.Worker{})).Return(nil)
```

File: Makefile (hunk 1 above)
Comment (prashanth26) [reply Makefile:130]: Done. Here - https://github.com/gardener/gardener-extension-provider-gcp/pull/183/commits/082dd6801beb95952bd04c5912f3d3d01497398e#diff-b67911656ef5d18c4ae36cb6741b7965R134

File: pkg/controller/worker/machines_test.go (hunk 2 above)
Comment (prashanth26) [reply pkg/controller/worker/machines_test.go]: You meant this right? - https://github.com/gardener/gardener-extension-provider-gcp/pull/183/commits/082dd6801beb95952bd04c5912f3d3d01497398e#diff-e61377d85357d3c316ed34aa40a57057R560

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**